api-get-object.go

```diff
@@ -132,8 +197,29 @@ type Object struct {
 	// Keeps track of closed call.
 	isClosed bool
 
+	// Keeps track of if this is the first call.
+	isStarted bool
+
 	// Previous error saved for future calls.
 	prevErr error
+
+	// Keeps track of if this object has been read yet.
+	beenRead bool
+}
+
+// setObjectInfo - blocks until the first request is completed and then either errors out
+// or sets the received objectInfo.
+func (o *Object) setObjectInfo() (int, error) {
+	firstResponse := <-o.firstResCh
+	// Error out here if the firstRequest caused an error.
+	if firstResponse.Error != nil {
+		return 0, firstResponse.Error
```

Comment:
This error should be set properly to 'o.prevErr' so that any subsequent operations on the same client object fail with the same error.